bool is_finished
uint8[50] message
